build/entrypoint.sh

```diff
@@ -9,18 +9,38 @@ if [ -f /podman_compose/kai/config.toml ]; then
   sed -i "s/^password =.*/password =\"$POSTGRES_PASSWORD\"/g" /kai/kai/config.toml
 fi
 
+until PGPASSWORD="$POSTGRES_PASSWORD" pg_isready -q -h kai_db -U $POSTGRES_USER -d $POSTGRES_DB ; do
+  sleep 1
+done
+
 TABLE=applications
-SQL_EXISTS=$(printf '\dt "%s"' "$TABLE")
-if [[ $(PGPASSWORD="$POSTGRES_PASSWORD" psql -h "kai_db" -U $POSTGRES_USER -d $POSTGRES_DB -c "$SQL_EXISTS") ]]
-then
-    echo "load-data has run already"
+SQL_EXISTS=$(printf "\dt %s" "$TABLE")
+STDERR="Did not find any relation"
+if PGPASSWORD="$POSTGRES_PASSWORD" psql -h kai_db -U $POSTGRES_USER -d $POSTGRES_DB -c "$SQL_EXISTS" 2>&1 | grep -q -v "$STDERR"; then
+  echo "################################################"
+  echo "load-data has run already run, starting server.#"
+  echo "################################################"
 else
-    echo "load-data has not run yet, starting ..."
-    echo "Fetching examples"
+  echo "################################################"
+  echo "load-data has never been run.                  #"
+  echo "Please wait, this will take a few minutes.     #"
+  echo "################################################"
+  sleep 5
+
+  if [ "$USE_HUB_IMPORTER" = "True" ]; then
+    cd /kai
+    python ./kai/hub_importer.py --loglevel ${LOGLEVEL} --config_filepath ./kai/config.toml ${IMPORTER_ARGS} ${HUB_URL}
```

Comment:
so this actually starts a long-running process that runs alongside the kai server, it won't terminate. In the compose it should probably be a separate container that also has access to the database. The hub importer watches the API indefinitely for analysis reports and adds them to the database as they occur.

Reply:
Understood. I will rework this. I think I have an idea of how to handle this.